### PR TITLE
Restructure into serapate apis

### PR DESCRIPTION
**Pull request by joswarmer**
Restructure to separate all different API's and total cleanup of all code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,9 +11,13 @@
     "url": "https://github.com/LionWeb-io/lionweb-repository/issues"
   },
   "scripts": {
-    "build": " npm run build -w packages",
-    "lint": "npm run lint -w packages",
+    "initialize": "npm run clean && npm install && npm run setup && npm run build",
+    "clean": "rm -rf node_modules/ && npm run clean -w packages --if-present",
```
Comment (ftomassetti) [inline package.json:15]: what does --if-present means? Does it run the task for all packages that have such task?

**Comment by ftomassetti** [comment]
Can we have a copy of `.prettierc` at the root level and refer to it from the single packages?

File: packages/bulkapi/src/controllers/BulkApi.ts [hunk 2]
```diff
@@ -2,48 +2,85 @@
...
+        res.send(result.queryResult)
+    }
+
+    createPartitions = async (req: Request, res: Response): Promise<void> => {
+        const chunk: LionWebJsonChunk = req.body
+        const validator = new LionWebValidator(chunk, getLanguageRegistry())
+        validator.validateSyntax()
+        validator.validateReferences()
+        if (validator.validationResult.hasErrors()) {
+            // console.log("STORE VALIDATION ERROR " + validator.validationResult.issues.map(issue => issue.errorMsg()))
```
Comment (ftomassetti) [inline packages/bulkapi/src/controllers/BulkApi.ts:41]: Can we remove this commented out code or alternatively use the logger?

File: packages/bulkapi/src/database/LionWebQueries.ts [hunk 3]
```diff
@@ -265,4 +296,4 @@ export function printMap(map: Map<string, string>): string {
     return result
 }
 
-export const LIONWEB_QUERIES = new LionWebQueries()
+// export const LIONWEB_QUERIES = new LionWebQueries()
```
Comment (ftomassetti) [inline packages/bulkapi/src/database/LionWebQueries.ts:299]: Should we remove this?

File: packages/test/src/test/RepositoryClient.ts [hunk 4]
```diff
@@ -29,62 +29,48 @@ export class RepositoryClient {
     }
 
     async testPartitions() {
-        console.log(`test.partitions`)
+        // console.log(`test.partitions`)
```
Comment (ftomassetti) [inline packages/test/src/test/RepositoryClient.ts:32]: Should remove this and the other commented out lines in this file?

File: packages/test/src/test/tests-issue19.ts [hunk 5]
```diff
@@ -16,12 +15,11 @@ describe("Repository tests", () => {
...
         it("test update single node", async () => {
             await storeFiles([
-                "./src/test/data/Disk_A.json",
-                "./src/test/data/add-new-nodes/Disk-add-new-nodes-single-node.json",
-                "./src/test/data/Disk_A.json",
+                "./data/Disk_A.json",
+                "./data/add-new-nodes/Disk-add-new-nodes-single-node.json",
+                "./data/Disk_A.json",
             ])
             // const afterRetrieve = await t.testRetrieve(["ID-2"])
```
Comment (ftomassetti) [inline packages/test/src/test/tests-issue19.ts:24]: Should we remove these lines?

**Review by ftomassetti — APPROVED** [review]
I am not a TS developer and the PR is huge, but I took a look for most obvious things.

It looks very good and a great improvement.

My only comment is that it could make sense to write a message on Slack when working on large refactorings to avoid others sending PRs at the same time to avoid conflicts

File: packages/additionalapi/src/controllers/AdditionalApi.ts [hunk 6]
```diff
@@ -1,27 +1,27 @@
...
+export class AdditionalApiImpl implements AdditionalApi {
+    constructor(private context: AdditionalApiContext) {
+    }
     /**
      * Get the tree with root `id`, for one single node
      * @param req
      * @param res
      */
-    async getNodeTree(req: Request, res: Response) {
+    getNodeTree = async (req: Request, res: Response)=> {
```
Comment (dslmeinte) [inline packages/additionalapi/src/controllers/AdditionalApi.ts:17]: Is not having a space before the `=>` intentional? Unfortunately, `prettier` doesn't seem to have an option for this – at least not specifically for double arrows.

Comment (joswarmer) [reply packages/additionalapi/src/controllers/AdditionalApi.ts:17]: No space is not intentional.

**Comment by joswarmer** [comment]
> Can we have a copy of `.prettierc` at the root level and refer to it from the single packages?

Prettier does not support this.

File: packages/bulkapi/src/controllers/BulkApi.ts (hunk 2 above)
Comment (joswarmer) [reply packages/bulkapi/src/controllers/BulkApi.ts:41]: Will remove it

File: packages/test/src/test/RepositoryClient.ts (hunk 4 above)
Comment (joswarmer) [reply packages/test/src/test/RepositoryClient.ts:32]: Yes

File: packages/additionalapi/src/controllers/AdditionalApi.ts (hunk 6 above)
Comment (joswarmer) [reply packages/additionalapi/src/controllers/AdditionalApi.ts:17]: Using an arrow function is intentional, as it will bind "this" correctly.

Comment (joswarmer) [reply packages/additionalapi/src/controllers/AdditionalApi.ts:17]: Websytorm show the following, making it almost impossible to see whether there is a space before the arrow:
image
The " : Promise<void> " is projected by the editor, it is not in the actual source code.

File: packages/test/src/test/tests-issue19.ts (hunk 5 above)
Comment (joswarmer) [reply packages/test/src/test/tests-issue19.ts:24]: Yes

File: packages/additionalapi/src/controllers/AdditionalApi.ts (hunk 6 above)
Comment (dslmeinte) [reply packages/additionalapi/src/controllers/AdditionalApi.ts]: Ah yes, and same thing (sometimes) with argument lists. But don't worry: this is just _my_ OCD ;)

**Pull request merged**